lib/commander.ts

```diff
@@ -57,21 +57,46 @@ Commander.prototype.getBuiltinCommands = function () {
  */
 Commander.prototype.createBuiltinCommand = function (commandName) {
   return {
-    string: generateFunction(commandName, "utf8"),
-    buffer: generateFunction(commandName, null),
+    string: generateFunction(null, commandName, "utf8"),
+    buffer: generateFunction(null, commandName, null),
   };
 };
 
+/**
+ * Create add builtin command
+ *
+ * @param {string} commandName - command name
+ * @return {object} functions
+ * @public
+ */
+Commander.prototype.addBuiltinCommand = function (commandName) {
+  Commander.prototype[commandName] = generateFunction(
+    commandName,
+    commandName,
+    "utf8"
+  );
+  Commander.prototype[commandName + "Buffer"] = generateFunction(
```

Comment:
I think https://github.com/luin/ioredis/blob/master/lib/pipeline.ts#L64 needs to be change to:
```typescript
Pipeline.prototype = Object.create(Commander.prototype);
```
for this to actually work.
e.g. https://repl.it/repls/DesertedCookedApplicationframework#index.js

Reply:
Why? 
The repl test will never work as you defining `newFn` after the prototype would have been assigned. 
Are you trying to keep the prototype in sync?

Reply:
It does work with (as it's using Object.create), that's what the repl test is demonstrating. Maybe it's clearer like this:
![image](https://user-images.githubusercontent.com/9082513/98117997-6bf9cb80-1ea2-11eb-8066-34fb71276daa.png)

Reply:
Ok, that's what I thought then. You use Object.create so that the prototype of the children gets new addition of the prototype of the parent.
Looks fine to me!